src/RequestParser.cpp

```diff
@@ -0,0 +1,551 @@
+	}
+	m_request.version.push_back(requestLine[i]);
+	return (requestLine.substr(++i));
+}
+
+/**
+ * @brief Checks the validity of an HTTP header field name.
+ *
+ * This function verifies if the provided HTTP header field name is valid
+ * according to HTTP specifications.
+ * It ensures that the header name does not end with whitespace and that all
+ * characters in the header name are valid according to `isValidHeaderFieldNameChar`.
+ *
+ * @param headerName The HTTP header field name to be validated.
+ * @throws std::runtime_error If the header name contains invalid characters or
+ *         ends with whitespace, an error is thrown and the error code is set to 400.
+ */
+void	RequestParser::checkHeaderName(const std::string& headerName)
+{
+	if (isspace(headerName[headerName.size() - 1])){
+		m_errorCode = 400;
+		throw std::runtime_error(ERR_HEADER_COLON_WHITESPACE);
+	}
+	for (size_t i = 0; i < headerName.size(); i++) {
+		if (!isValidHeaderFieldNameChar(headerName[i])) {
+			m_errorCode = 400;
+			throw std::runtime_error(ERR_HEADER_NAME_INVALID_CHAR);
+	 	}
+	}
+}
+
+/**
+ * @brief Checks the validity of the Content-Length header field.
+ *
+ * This function verifies the validity of the Content-Length header field in an HTTP request.
+ * It ensures that:
+ * - There is only one Content-Length header or multiple with the same value.
+ * - The value of Content-Length is a valid numeric string.
+ * If the header is invalid or there are multiple Content-Length headers with conflicting values,
+ * an appropriate error code is set, and a `std::runtime_error` is thrown.
+ *
+ * @param headerName The name of the HTTP header field.
+ * @param headerValue The value associated with the HTTP header field.
+ * @throws std::runtime_error If there are multiple conflicting Content-Length
+ *         headers or if the Content-Length value is invalid, an error is thrown
+ *         and the error code is set to 400.
+ */
+void	RequestParser::checkContentLength(const std::string& headerName, std::string& headerValue)
+{
+	if (headerName == "Content-Length") {
+		if (headerValue.empty()) {
+			m_errorCode = 400;
+			throw std::runtime_error(ERR_INVALID_CONTENT_LENGTH);
+		}
+		if (m_request.headers.find("Content-Length") != m_request.headers.end()
+			&& m_request.headers["Content-Length"] != headerValue) {
+				m_errorCode = 400;
+				throw std::runtime_error(ERR_MULTIPLE_CONTENT_LENGTH_VALUES);
+		}
+
+		std::vector<std::string> strValues = split(headerValue, ',');
+		std::vector<double>	numValues;
+		for (size_t i = 0; i < strValues.size(); i++) {
+			char 	*endptr;
+			double	contentLength = strtod(strValues[i].c_str(), &endptr);
+			if (!contentLength || *endptr != '\0') {
+				m_errorCode = 400;
+				throw std::runtime_error(ERR_INVALID_CONTENT_LENGTH);
+			}
+			numValues.push_back(contentLength);
+			if (i != 0 && contentLength != numValues[i - 1]) {
+				m_errorCode = 400;
+				throw std::runtime_error(ERR_MULTIPLE_CONTENT_LENGTH_VALUES);
+			}
+		}
```

Comment:
Now I get it: I just read, that the form ``"Content-Length: 42, 42"`` may be accepted as exception: [LINK](https://www.rfc-editor.org/rfc/rfc9110.html#section-8.6).

I think using strtol instead of strtod would make more sense - a 4.6 would be parsed I think
Also again: I think the split is a bit harsh here, iterating over the string would be less costly

Reply:
Will replace with strtol. 
I prefer this variant as it is logically precise and I don't think in nowadays environments it makes much of a difference performance wise.

Reply:
Small things add up, but I'm ok with it